### PR TITLE
AP_InertialSensor: added support for more ADIS IMUs

### PR DESCRIPTION
**Pull request by tridge**
support 32 bit delta angles and velocities
tested on a CUAV-X7 with ADIS16470 and an external ADIS16507 with 32-bit delta-angles and delta-velocity

### CODE REVIEW

**Review by andyp1per — APPROVED** [review]

File: libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp [hunk 1]
```diff
@@ -305,6 +305,134 @@ void AP_InertialSensor_Backend::_notify_new_gyro_raw_sample(uint8_t instance,
...
+{
+    if ((1U<<instance) & _imu.imu_kill_mask) {
+        return;
+    }
+    float dt;
+
+    _update_sensor_rate(_imu._sample_gyro_count[instance], _imu._sample_gyro_start_us[instance],
+                        _imu._gyro_raw_sample_rates[instance]);
+
+#if 0
```
Comment (andyp1per) [inline libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp:322]: debug flag?

**Review by andyp1per — CHANGES_REQUESTED** [review]

File: libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp:322]: fixed

File: libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp [hunk 2]
```diff
@@ -428,6 +556,85 @@ void AP_InertialSensor_Backend::_notify_new_accel_raw_sample(uint8_t instance,
...
+{
+    if ((1U<<instance) & _imu.imu_kill_mask) {
+        return;
+    }
+    float dt;
+
+    _update_sensor_rate(_imu._sample_accel_count[instance], _imu._sample_accel_start_us[instance],
+                        _imu._accel_raw_sample_rates[instance]);
+
+#if 1
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp]: fixed, thanks

File: libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp [hunk 3]
```diff
@@ -305,6 +305,134 @@ void AP_InertialSensor_Backend::_notify_new_gyro_raw_sample(uint8_t instance,
     }
 }
 
+/*
+  handle a delta-angle sample from the backend. This assumes FIFO style sampling and
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp:311]: why notch changes? The new delta_angle function does handle the notch calls

File: libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp [hunk 4]
```diff
@@ -428,6 +556,85 @@ void AP_InertialSensor_Backend::_notify_new_accel_raw_sample(uint8_t instance,
     }
 }
 
+/*
+  handle a delta-velocity sample from the backend. This assumes FIFO style sampling and
+  the sample should not be rotated or corrected for offsets
+ */
```
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp:560]: done

File: libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp]: fixed

File: libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (tridge) [reply libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp]: fixed

**Review by peterbarker — APPROVED** [review]

**Pull request merged**